src/export/extensionAPI.ts

```diff
@@ -0,0 +1,72 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+// These APIs are for more advanced usages
+// For example extend charts and components, creating graphic elements, formatting.
+import ComponentModel, { ComponentModelConstructor } from '../model/Component';
```

Comment:
Will this `extensionAPI` file name being confused with the `echarts/src/core/ExtensionAPI` ?

And I'm not sure about the idea of providing same APIs in `export/api` and `export/extensionHelper`.

Reply:
> Will this extensionAPI file name being confused with the echarts/src/core/ExtensionAPI ?

I've noticed this naming issue. But I don't know why `ExtensionAPI` didn't get a name like `EChartsAPI`.

> And I'm not sure about the idea of providing same APIs in `export/api` and `export/extensionHelper`.

It seems `export/api` is not public for users since it isn't in the final distributions. But `echarts/core` imports some parts from
it. I think these APIs and types are more often to be used for extension developers, so extract them to the `extensionHelper`. It may look more intuitive from the naming.

I don't have a great idea about this design. It may need further discussion.

Reply:
> But I don't know why ExtensionAPI didn't get a name like EChartsAPI or EChartsInstanceAPI.
That's a good point. The original design is every component and chart can be treated as an extension. The `ExtensionAPI` is provided to these components and charts. But now we can see it seems to be not a good idea. `EChartsAPI` should be better here.

> It seems export/api is not public for users since it isn't in the final distributions. But echarts/core imports some parts from it. I think these APIs and types are more often to be used for extension developers, so extract them to the extensionHelper. It may look more intuitive from the naming.

What I thought is extension developers can still use the methods exposed from `echarts/core`. Or could we re-export `export/api` in the `export/extensionHelper` directly?